src/store/CategoricalDisplayStore.ts

```diff
@@ -360,16 +369,24 @@ export default class CategoricalDisplayStore {
     handleTitleClick(categoryId: string): void {
         this.activeCategories = [];
         for (const item of this.filteredItems) {
-            if (item.categoryId === categoryId) {
-                item.active = !item.active;
-
-                if (item.active) {
-                    setTimeout(() => this.handleFilterClick(categoryId), 250);
-                }
-            }
-
+            const isItemActive = !item.hasSubgroup ? item.categoryId === categoryId : item.subgroups?.filter((subgroup: TCategorizedSymbolItem) => subgroup.categoryId === categoryId).length > 0;
+            if (isItemActive) {
+                if (item.hasSubgroup){
+                    const triggered_subgroup = item.subgroups.find((subgroup: TCategorizedSymbolItem) => subgroup.categoryId === categoryId);
+
+                    if (triggered_subgroup != undefined) {
+                        triggered_subgroup.active = !triggered_subgroup.active;
+                    }
+                        setTimeout(() => this.handleFilterClick(categoryId), 250);
+                } else {
+                    item.active = !item.active;
+    
+                    if (item.active) {
+                        setTimeout(() => this.handleFilterClick(categoryId), 250);
+                    }
```

Comment:
```suggestion    
                    if (!item.active) {
                        setTimeout(() => this.handleFilterClick(categoryId), 250);
                    }
```

Reply:
We still need to toggle `item.active` regardless of the condition, so it won't work if I remove it